src/test/java/io/vertx/test/core/DeploymentTest.java

```diff
@@ -987,6 +987,46 @@ public void testUndeployAllNoDeployments() throws Exception {
     vertx = null;
   }
 
+  @Test
+  public void testGetInstanceCount() throws Exception {
+    class MultiInstanceVerticle extends AbstractVerticle {
+      @Override
+      public void start() {
+        assertEquals(vertx.getOrCreateContext().getInstanceCount(), 1);
+      }
+    }
+
+    vertx.deployVerticle(new MultiInstanceVerticle(), ar -> {
+      assertTrue(ar.succeeded());
+      testComplete();
+    });
+    await();
+    Deployment deployment = ((VertxInternal) vertx).getDeployment(vertx.deploymentIDs().iterator().next());
+    vertx.undeploy(deployment.deploymentID());
+  }
+
+  @Test
+  public void testGetInstanceCountMultipleVerticles() throws Exception {
+    AtomicInteger messageCount = new AtomicInteger(0);
+    AtomicInteger totalReportedInstances = new AtomicInteger(0);
+
+    vertx.eventBus().consumer("instanceCount", event -> {
+      messageCount.incrementAndGet();
+      totalReportedInstances.addAndGet((int)event.body());
+      if(messageCount.intValue() == 3) {
+        assertEquals(9, totalReportedInstances.get());
+      }
+    });
+
+    vertx.deployVerticle(TestVerticle3.class.getCanonicalName(), new DeploymentOptions().setInstances(3), ar -> {
+      assertTrue(ar.succeeded());
```

Comment:
Not sure this test is valid as the deployment handler could be called before the message consumer has got all the instanceCount messages.

Better to put the testComplete() in the message handler.